src/main/kotlin/no/nav/klage/oppgave/service/distribusjon/KlagebehandlingDistribusjonService.kt

```diff
@@ -0,0 +1,76 @@
+package no.nav.klage.oppgave.service.distribusjon
+
+import no.nav.klage.oppgave.domain.klage.BrevMottaker
+import no.nav.klage.oppgave.domain.klage.Vedtak
+import no.nav.klage.oppgave.service.KlagebehandlingService
+import no.nav.klage.oppgave.util.getLogger
+import no.nav.klage.oppgave.util.getSecureLogger
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.*
+
+@Service
+@Transactional
+class KlagebehandlingDistribusjonService(
+    private val klagebehandlingService: KlagebehandlingService,
+    private val vedtakDistribusjonService: VedtakDistribusjonService,
+    private val klagebehandlingAvslutningService: KlagebehandlingAvslutningService
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+        private val secureLogger = getSecureLogger()
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
```

Comment:
Jeg leste om NESTED nå, og tenker den gjør det vi vil. Den trenger heller ikke `JTATransactionManager (som REQUIRES_NEW krever?)` som ikke tror vi kanskje ikke har nå?

Reply:
Jeg skrev det om jeg nå, til noe jeg har mer tro på. Det er kanskje tregere, men det tror jeg ikke gjør så mye. Jeg kjører uten transaksjoner til å begynne med, og så lager jeg en ny transaksjon (helt vanlig REQUIRED) for hvert steg, og henter opp Klagebehandlingen på nytt for hver gang. Det bør både fungere mtp optimistic locking, og det bør sørge for at vi flusher stegene etterhvert som vi gjør dem.